docs/specification/draft/client/elicitation.mdx

```diff
@@ -317,16 +521,199 @@ The three response actions are:
+3. Clients **SHOULD** allow users to reject elicitation requests at any time
+4. Clients **SHOULD** implement rate limiting
+5. Clients **SHOULD** present elicitation requests in a way that makes it clear what information is being requested and why
+6. Servers **MUST** bind elicitation requests to the user's identity
+
+### Safe URL Handling
+
+Clients implementing out-of-band elicitation **MUST** handle URLs carefully to prevent users from unknowingly clicking malicious links.
+
+1. Servers **MUST NOT** include URLs in any message or schema fields as part of a form elicitation request.
+2. Servers **MUST NOT** include URLs in any message or schema fields as part of an out-of-band elicitation request, except for the `url` field.
+
+These requirements ensure that client implementations have clear rules about when to present a URL to the user, so that other rules about user consent and SSRF protection (below) can be consistently applied.
+
+### Identifying the User
+
+Servers **MUST NOT** rely on client-provided user identification, as this can be forged.
+Instead, servers **SHOULD** follow [security best practices](/specification/draft/basic/security_best_practices).
+
+Non-normative examples:
+
+- Incorrect: Treat user input like "I am joe@example.com" as authoritative
+- Correct: Rely on the [MCP authorization server](/specification/draft/basic/authorization) to identify the user
+
+### Form Mode Security
+
+1. Servers **MUST NOT** request sensitive information (passwords, API keys, etc.) via form mode
+2. Clients **SHOULD** validate all responses against the provided schema
+3. Servers **SHOULD** validate received data matches the requested schema
+
+### Out-of-Band Mode Security
+
+Clients **MUST NOT** open a user agent (e.g. browser) without explicit consent from the user
+
+#### Server-Side Request Forgery (SSRF)
+
+Since clients facilitate the opening of URLs provided by servers, they **MUST** implement SSRF protections, including:
+
+- Requiring the `https://` scheme for all out-of-band URLs (no HTTP, `file://`, etc.)
+- Blocking requests to internal IP ranges (e.g., `127.0.0.1`, `10.0.0.0/8`, etc.)
+- Clearly rendering or distinguishing Unicode characters (e.g. punycode URLs) to avoid "look-alike" misdirections
+- Clearly communicating the destination server and target URL to the user when asking for consent
+
+Further recommendations can be found in the OWASP [SSRF Prevention Cheat Sheet](https://cheatsheetseries.owasp.org/cheatsheets/Server_Side_Request_Forgery_Prevention_Cheat_Sheet.html).
+
+#### Phishing
+
+Out-of-band elicitation returns a URL that an attacker can use to send to a victim. The MCP Server **MUST** verify the identity of the user who opens the URL before accepting information.
+
+Typically identity verification is done by leveraging the [MCP authorization server](/specification/draft/basic/authorization) to identify the user, through a session cookie or equivalent in the browser.
```

Comment:
OAuth is definitely a route, but a simple alternative for servers to implement may be to include a secret, random string in a URL fragment or query parameter. OAuth checks only work for HTTP servers.

Reply:
The problem with that approach is that the URL becomes a pre-authenticated URL or "magic link", which does not actually prevent the kind of phishing attack discussed here.